src/api/api.test.tsx

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import axios from 'axios';
+import { postSecret, getSecret } from './api';
+
+test('should fire an api post request', async () => {
+  // Arrange
+  const apirequest = jest.spyOn(axios, 'post').mockImplementationOnce(async () => Promise.resolve({ data: {} }));
+  // Act
+  await postSecret({ secret: 'test message' });
+  // Assert
+  expect(apirequest).toHaveBeenCalledTimes(1);
+  apirequest.mockClear();
+});
+
+test('should return a promise from post request', async () => {
+  // Arrange
+  jest.spyOn(axios, 'post').mockImplementationOnce(async () => Promise.resolve({ data: {} }));
+  // Act
+  const mockPromise = await postSecret({ secret: 'test message' });
+  // Assert
+  expect(mockPromise).toEqual(await Promise.resolve({ data: {} }));
+});
+
+test('should fetch data with secret id from post request', async () => {
+  // Arrange
+  const id = { id: '1234567' };
+  const resp = { data: id };
+  jest.spyOn(axios, 'post').mockImplementationOnce(async () => Promise.resolve(resp));
+  // Act
+  const expRes = await postSecret({ secret: 'test message' });
+  // Assert
+  expect(expRes.data.id).toEqual(await Promise.resolve('1234567'));
```

Comment:
Why do we still need the Promise here?
```suggestion
  expect(expRes.data.id).toEqual('1234567');
```

Reply:
Yes, I find this right after committing.
Can I ask if you have seen my previous comments? I mentioned this and also asked some other questions there.

Reply:
Cool! The questions are being either answered here or the meeting.